test/framework/event-wrapper.ts

```diff
@@ -0,0 +1,329 @@
+import {expect} from 'chai'
+import {
+    Contract,
+    ContractReceipt,
+    ContractTransaction,
+    Event,
+    EventFilter,
+    utils
+} from 'ethers'
+import {TypedEvent, TypedEventFilter} from '../../typechain-types/common'
+import {EventListener} from './event-listener'
+
+function eventsArg(
+    name: string,
+    receipt: ContractReceipt,
+    emitter?: Contract
+): unknown[] {
+    const found: unknown[] = []
+
+    // eslint-disable-next-line no-undefined
+    if (emitter === undefined) {
+        for (const entry of receipt.events ?? []) {
+            if (entry?.event === name) {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                found.push(entry.args!)
+            }
+        }
+    } else {
+        const addr = emitter.address.toUpperCase()
+        const parser = emitter.interface
+
+        const fragment = parser.getEvent(name)
+        const id = utils.id(fragment.format())
+
+        for (const entry of receipt.logs) {
+            if (
+                entry.topics[0] === id &&
+                entry.address.toUpperCase() === addr
+            ) {
+                const parsed = parser.decodeEventLog(
+                    fragment,
+                    entry.data,
+                    entry.topics
+                )
+                found.push(parsed)
+            }
+        }
+    }
+
+    expect(
+        found.length,
+        `Failed to find any event matching name: ${name}`
+    ).is.greaterThan(0)
+
+    return found
+}
+
+export type ContractReceiptSource =
+    | ContractReceipt
+    | Promise<ContractReceipt>
+    | ContractTransaction
+    | Promise<ContractTransaction>
+
+export interface EventFactory<TArgObj = unknown> {
```

Comment:
suggestion; s/`TArgObj`/`T`

This applies for instances of `TArgObj`. `T` being a more common used for generics

Reply:
yeah, it was the first ugly version - will fix